### PR TITLE
fix(v-model): v-model ended with spaces

### LINKED ISSUE #7730: Problem when a v-model end with spaces
### Version
2.5.13

### Reproduction link
[https://codepen.io/leocardoso/pen/OQqNYz?editors=1011](https://codepen.io/leocardoso/pen/OQqNYz?editors=1011)

### Steps to reproduce
Create an input with a v-model like  <input v-model="obj.value " />

And try to change its value

### What is expected?
The value of model change

### What is actually happening?
The value of model doesn't change and it creates a new property to the Object ended with space like obj['value ']

### PR DESCRIPTION
**Pull request by Leocardoso94**
fix #7730

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: src/compiler/directives/model.js [hunk 1]
```diff
@@ -69,6 +69,7 @@ type ModelParseResult = {
 
 export function parseModel (val: string): ModelParseResult {
   len = val.length
+  val = val.trim()
```
Comment (posva) [inline src/compiler/directives/model.js]: Shouldn't this be above the previous line 🤔

Comment (Leocardoso94) [reply src/compiler/directives/model.js]: It's true, I fixed this in the last commit

File: test/unit/features/directives/model-text.spec.js [hunk 2]
```diff
@@ -19,6 +19,17 @@ describe('Directive v-model text', () => {
     }).then(done)
   })
 
+  it('should not create a new property ended with spaces when', () => {
```
Comment (posva) [inline test/unit/features/directives/model-text.spec.js]: Could you add a test that fails without the fix above please? This one passes
Also, could you rename the test title to `should work with extra whitespace properties` or something similar, please?

Comment (Leocardoso94) [reply test/unit/features/directives/model-text.spec.js]: I did not understand the part of the test. Do I create another pull request with only the test? Because the current test fails without this fix

**Comment by posva** [comment]
The current test does pass without the patch you added but it should fail. I'm asking if you could add a test case that fails without the `val = val.trim()` but passes with it

**Comment by Leocardoso94** [comment]
Hi @posva , I add a e2e that fails without the `val.trim()`

Apparently, in the unit test, it was not possible to simulate this case, I believe it only creates the new property with extra spaces only when it is typed in the browser.

Q. The extra property is only created if you pass a v-model from an object like  `v-model="obj.value "`

**Comment by posva** [comment]
You needed to add a test with a nested object 😄 
I added the test at #7737 and kept your initial commits
